deployment/terraform/dump.go

```diff
@@ -257,12 +252,36 @@ func (t *Terraform) executeDatabaseCommands(extraCommands []deployment.Cmd) erro
 func (t *Terraform) executeCommands(commands []deployment.Cmd) error {
 	for _, c := range commands {
 		mlog.Info(c.Msg)
+
+		errors := make(chan error, len(c.Clients))
+		wg := sync.WaitGroup{}
+
 		for _, client := range c.Clients {
-			mlog.Debug("Running cmd", mlog.String("cmd", c.Value))
-			if out, err := client.RunCommand(c.Value); err != nil {
-				return fmt.Errorf("failed to run cmd %q: %w %s", c.Value, err, out)
-			}
+			wg.Add(1)
+			go func() {
+				defer wg.Done()
+				mlog.Debug("Running cmd", mlog.String("cmd", c.Value), mlog.String("ip", client.IP))
+				if out, err := client.RunCommand(c.Value); err != nil {
+					errors <- fmt.Errorf("failed to run cmd %q on %s: %w %s", c.Value, client.IP, err, out)
+				}
+			}()
 		}
+
+		wg.Wait()
+		go func() {
+			close(errors)
+		}()
```

Comment:
Why do we run this in a goroutine?

Reply:
These are the remains of me doing more complex code with a `doneChan` and all that jazz, I simplified it in this way to at least maintain the "the receiver should not close the channel" mantra. Should work either way I guess.

Reply:
It would work either way, indeed. I'd prefer to remove the goroutine myself, but will leave it up to you :)